### ISSUE TITLE
 Cursor turns to Arrow on some sections the IOU preview component

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#19017]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Sign in with account A
2. Send an IOU to account B
3. Shit to the tab where account B is signed in
4. Open the sent IOU
5. Hover over all parts of the IOU preview component

## Expected Result: 
Cursor shouldn't turn to Arrow since the component has hover effect and the whole preview component should be clickable
## Actual Result: 
Curosr turns to Arrow on some section of the preview component

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.14-9
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/5e8e9b00-47ed-4e82-a7cc-bb98c0ad949e

https://github.com/Expensify/App/assets/43996225/c2c8dcfc-5742-4f74-8ede-8955758b22cc

**Expensify/Expensify Issue URL:**
**Issue reported by:** @natnael-guchima
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1684137959661979

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by therealsujitk** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

The `Pressable` is taking the space of the entire action item instead of just the `IOUPreview` component.

### What is the root cause of that problem?

This is default behaviour of react native, components (In this case `Pressable`) expand to fill it's parent as react native follows a flexbox system to style components.

### What changes do you think we should make in order to solve the problem?


To resolve this, we simply apply the `IOUPreview` component styles to the `Pressable` instead of the `View`.

### What alternative solutions did you explore? (Optional)
None.

**Comment by conorpendergrast** [comment]
Reproduced now, thanks!

**Comment by conorpendergrast** [comment]
@rushatgabhane there's a speedy proposal [here](https://github.com/Expensify/App/issues/19017#issuecomment-1549907508), if you could give that a review!

**Comment by conorpendergrast** [comment]
@rushatgabhane Bump on reviewing [that proposal](https://github.com/Expensify/App/issues/19017#issuecomment-1549907508), please

**Comment by papistacoding** [comment]
Hey, I have a solution to this issue. I submitted a proposal to upwork.
Regards,
Bruno

**Comment by papistacoding** [comment]
Hey, I have solution to this issue. 
Upwork proposal is submitted
Regards,
Bruno

**Comment by conorpendergrast** [comment]
@papistacoding Hey! We only accept proposals on the github issue (right where you are!), the format [you see above please](https://github.com/Expensify/App/issues/19017#issuecomment-1549907508). Thanks!

**Comment by papistacoding** [comment]
@conorpendergrast Thanks for info. So proposal from @therealsujitk has been already approved?

**Comment by conorpendergrast** [comment]
Not so far! So there's still an opportunity to post a fresh proposal here in that format, that @rushatgabhane can then review

**Comment by papistacoding** [comment]
@conorpendergrast sorry to ask so many questions. I got the resolution working as intended, but how do I earn money on upwork? First I put proposal here? please tag me in reply so I get a notification

**Comment by conorpendergrast** [comment]
@papistacoding I'd recommend taking a look at [this guide](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#payment-for-contributions), and then asking any questions in the [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) Slack channel!

**Comment by papistacoding** [comment]
**Proposal**
Please re-state the problem that we are trying to solve in this issue.
The Pressable is taking the space of the entire action item instead of just the IOUPreview component.

What is the root cause of that problem?
This is default behaviour of react native, components (In this case Pressable) expand to fill it's parent as react native follows a flexbox system to style components.

What changes do you think we should make in order to solve the problem?
To resolve this, we add max-width to pressable component and we add the margin of mt2. we remove margin from inner element which comes from props. we remove unnecessary props.

1. adding maxWidth and margin:
    return (
        <Pressable
            onPress={props.onPreviewPressed}
            onPressIn={() => DeviceCapabilities.canUseTouchScreen() && ControlSelection.block()}
            onPressOut={() => ControlSelection.unblock()}
            onLongPress={showContextMenu}
            style={{maxWidth: variables.sideBarWidth, ...styles.mt2}}
        >
            {childContainer}
        </Pressable>
    );
   
2. removing props from MoneyRequestAction.js:

const propTypes = {
    /** All the data of the action */
    action: PropTypes.shape(reportActionPropTypes).isRequired,

    /** The ID of the associated chatReport */
    chatReportID: PropTypes.string.isRequired,

    /** The ID of the associated request report */
    requestReportID: PropTypes.string.isRequired,

    /** Is this IOUACTION the most recent? */
    isMostRecentIOUReportAction: PropTypes.bool.isRequired,

    /** Popover context menu anchor, used for showing context menu */
    contextMenuAnchor: refPropTypes,

    /** Callback for updating context menu active state, used for showing context menu */
    checkIfContextMenuActive: PropTypes.func,

    /* Onyx Props */
    /** chatReport associated with iouReport */
    chatReport: PropTypes.shape({
        /** The participants of this report */
        participants: PropTypes.arrayOf(PropTypes.string),

        /** Whether the chat report has an outstanding IOU */
        hasOutstandingIOU: PropTypes.bool.isRequired,
    }),

    /** IOU report data object */
    iouReport: iouReportPropTypes,

    /** Array of report actions for this report */
    reportActions: PropTypes.objectOf(PropTypes.shape(reportActionPropTypes)),

    /** Whether the IOU is hovered so we can modify its style */
    isHovered: PropTypes.bool,

    network: networkPropTypes.isRequired,

    /** Session info for the currently logged in user. */
    session: PropTypes.shape({
        /** Currently logged in user email */
        email: PropTypes.string,
    }),

    ...withLocalizePropTypes,
};

const defaultProps = {
    contextMenuAnchor: undefined,
    checkIfContextMenuActive: () => {},
    chatReport: {
        participants: [],
    },
    iouReport: {},
    reportActions: {},
    isHovered: false,
    session: {
        email: null,
    },
};


 return (
        <>
            <IOUPreview
                iouReportID={props.requestReportID}
                chatReportID={props.chatReportID}
                isBillSplit={hasMultipleParticipants}
                action={props.action}
                contextMenuAnchor={props.contextMenuAnchor}
                checkIfContextMenuActive={props.checkIfContextMenuActive}
                shouldShowPendingConversionMessage={shouldShowPendingConversionMessage}
                onPreviewPressed={onIOUPreviewPressed}
                containerStyles={[styles.cursorPointer, props.isHovered ? styles.iouPreviewBoxHover : undefined]}
                isHovered={props.isHovered}
            />
        </>
    );
    
    this resolves the issue completely and we keep the structure of IOUPreview untouched

**Comment by therealsujitk** [comment]
Hey @papistacoding :wave: you'll have to properly format your proposal using Markdown and your code should go in code blocks with syntax highlighting otherwise your proposal just becomes unreadable and reviewers will probably just ignore it.

The proposal's markdown code can be obtained [here](https://github.com/Expensify/App/blob/main/contributingGuides/PROPOSAL_TEMPLATE.md?plain=1).

**Comment by conorpendergrast** [comment]
@rushatgabhane Two proposals for your to review here!

**Comment by therealsujitk** [comment]
I found another issue with the same RCA - #19717. I think both of them can be fixed together.

**Comment by therealsujitk** [comment]
@hoangzinh Now that you mention it, I re-read the issue and I might have been wrong about the expected result. However personally I think we should change it to **1**. There were several cases where I clicked on **2** just to switch to the browser window (if another window was open on top) or to lose focus from the composer and it took me to the IOU report which *really* annoyed me.

**Comment by therealsujitk** [comment]
@hoangzinh Now that you mention it, I re-read the issue and I might have been wrong about the expected result. However personally I think we should change it to **1**. There were several cases where I clicked on **2** just to switch to the browser window (if another window was open on top) or to lose focus from the composer and it took me to the IOU report which *really* annoyed me.

@conorpendergrast @yuwenmemon @rushatgabhane can you confirm what the expected result should be?

**Comment by Natnael-Guchima** [comment]
@hoangzinh sorry for my late reply.  I will take a look at this tomorrow and I will late you know.

**Comment by Natnael-Guchima** [comment]
@hoangzinh Yes, it doesn't seem you can't show the pay button and the IOU preview components together - they seem to be separated on the latest releases. The cursor issue persists on both of the separated components though. 


https://github.com/Expensify/App/assets/57330233/f52125fb-b00c-4b3a-8e12-299b6a25acff

**Comment by greg-schroeder** [comment]
@rushatgabhane @yuwenmemon - I have this related issue: https://github.com/Expensify/App/issues/19717 - do you think that should be folded into this one or handled separately?

**Comment by therealsujitk** [comment]
@greg-schroeder can you confirm the expected result as mentioned in [this comment](https://github.com/Expensify/App/issues/19017#issuecomment-1565484791) as well (The OP seems to say **2** is the expected result but I just want to be sure), depending on that the two issues may or may not be related and I may have to rewrite my proposal.

**Comment by greg-schroeder** [comment]
I don't know which approach is correct, so I'll wait on @yuwenmemon or @rushatgabhane to weigh in

**Comment by therealsujitk** [comment]
> Cursor turns to `hand`  only in this red section, and it's clickable only in the red section.

I believe this is already the case (though I personally don't like this behaviour), can you check with staging and confirm?

**Comment by conorpendergrast** [comment]
@yuwenmemon What's your take here? I'm erring on agreeing with @therealsujitk that this should actually be a feature request, not a bug

**Comment by yuwenmemon** [comment]
Agreed! :+1:

**Comment by conorpendergrast** [comment]
Ok, closing as a feature request, not a bug!
